### PR TITLE
Sphinx: Enable nitpicky option

### PR DESCRIPTION
**Pull request by jschueller**
allows to check internal links
also fixes some external links using sphinx linkchecker target (not run by default because of timeouts on CI)

### CODE REVIEW

File: python/doc/examples/functional_modeling/vectorial_functions/plot_createMultivariateFunction.py [hunk 1]
```diff
@@ -106,8 +105,7 @@
 # In that case, the univariate polynomials are orthogonal with respect to a measure :math:`\mu`.
 #
 # We have the family of polynomials: :math:`(x_i \mapsto \phi^i_k(x_i))_{k \geq 0}`
-# for :math:`1 \leq i \leq d`. The :meth:`enumerate` function stores of the
-# :class:`~openturns.LinearEnumerateFunction` class the way
+# for :math:`1 \leq i \leq d`. The class :class:`~openturns.EnumerateFunction` decides the way
```
Comment (josephmure) [inline python/doc/examples/functional_modeling/vectorial_functions/plot_createMultivariateFunction.py:108]:
```suggestion
# for :math:`1 \leq i \leq d`. The :class:`~openturns.EnumerateFunction` class decides the way
```

File: python/doc/examples/reliability_sensitivity/design_of_experiments/plot_smolyak_indices.py [hunk 2]
```diff
@@ -33,10 +33,10 @@
 # In the first example, we print the indices involved in
 # Smolyak-Legendre quadrature of level 3.
 # The multi-indices are computed using the
-# :meth:`~openturns.SmolyakExperiment.computeCombination` method.
+# :py:meth:`openturns.experimental.SmolyakExperiment.computeCombination` method.
```
Comment (josephmure) [inline python/doc/examples/reliability_sensitivity/design_of_experiments/plot_smolyak_indices.py:36]: Why is "py" necessary (here and at other places)?

File: python/doc/theory/meta_modeling/kriging.rst [hunk 3]
```diff
@@ -135,7 +135,7 @@ Kriging may also be referred to as *Gaussian process regression*.
 .. topic:: References:
 
     - [dubourg2011]_
-    - S. Lophaven, H. Nielsen and J. Sondergaard, 2002, "DACE, A Matlab kriging toolbox", Technichal University of Denmark. http://www2.imm.dtu.dk/projects/dace/
+    - S. Lophaven, H. Nielsen and J. Sondergaard, 2002, "DACE, A Matlab kriging toolbox", Technichal University of Denmark.
```
Comment (josephmure) [inline python/doc/theory/meta_modeling/kriging.rst:138]:
```suggestion
    - S. Lophaven, H. Nielsen and J. Sondergaard, 2002, "DACE, A Matlab kriging toolbox", Technichal University of Denmark. https://www.omicron.dk/dace/dace.pdf
```
This URL works.

File: python/doc/user_manual/base_objects.rst [hunk 4]
```diff
@@ -94,10 +101,21 @@ Vectors
...
     Indices
     IndicesCollection
+    UnsignedIntegerCollection
     Point
     PointWithDescription
     ComplexCollection
     ScalarCollection
+    FunctionCollection
+    PolynomialCollection
+    OptimizationResultCollection
```
Comment (josephmure) [inline python/doc/user_manual/base_objects.rst:111]: About all these collections: maybe we should revise the doc of the "Collection" page to remove all the examples and simply say these are underlying C++ objects which should not be directly used from the Python API.

File: python/src/Bonmin_doc.i.in [hunk 5]
```diff
@@ -39,17 +39,21 @@ Notes
 
 *Algorithms parameters:*
 
-  Bonmin algorithms can be adapted using numerous parameters, described  `here <https://www.coin-or.org/Bonmin/options_list.html#sec:options_list>`_. These parameters can be modified using the :class:`~openturns.ResourceMap`. For every option ``optionName``, one simply add a key named ``Bonmin-optionName`` with the value to use, as shown below::
+  Bonmin algorithms can be adapted using numerous parameters, described  `here <https://www.coin-or.org/Bonmin/options_list.html>`_.
+  These parameters can be modified using the :class:`~openturns.ResourceMap`.
+  For every option ``optionName``, one simply add a key named ``Bonmin-optionName`` with the value to use, as shown below::
```
Comment (josephmure) [inline python/src/Bonmin_doc.i.in:44]:
```suggestion
  For every option ``optionName``, simply add a key named ``Bonmin-optionName`` with the value to use, as shown below::
```

File: python/src/DistributionImplementation_doc.i.in [hunk 6]
```diff
@@ -821,7 +821,9 @@ Only valid for univariate and bivariate distributions.
 
 See Also
 --------
-computeCDF, viewer.View, ResourceMap
+openturns.Distribution.computeCDF
+openturns.viewer.View
+openturns.ResourceMap
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:826]: I would completely remove the "See also" section here.

File: python/src/DistributionImplementation_doc.i.in [hunk 7]
```diff
@@ -869,7 +871,10 @@ graph : :class:`~openturns.Graph`
 
 See Also
 --------
-computeCDF, getMarginal, viewer.View, ResourceMap
+openturns.Distribution.computeCDF
+openturns.Distribution.getMarginal
+openturns.viewer.View
+openturns.ResourceMap
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:877]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 8]
```diff
@@ -1302,7 +1307,8 @@ graph : :class:`~openturns.Graph`
 
 See Also
 --------
-computePDF, getMarginal, viewer.View, ResourceMap
+openturns.Distribution.computePDF
+openturns.Distribution.getMarginal
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:1311]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 9]
```diff
@@ -1341,7 +1347,8 @@ graph : :class:`~openturns.Graph`
 
 See Also
 --------
-computeLogPDF, getMarginal, viewer.View, ResourceMap
+openturns.Distribution.computeLogPDF
+openturns.Distribution.getMarginal
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:1351]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 10]
```diff
@@ -1385,7 +1392,8 @@ graph : :class:`~openturns.Graph`
 
 See Also
 --------
-computeCDF, getMarginal, viewer.View, ResourceMap
+openturns.Distribution.computeCDF
+openturns.Distribution.getMarginal
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:1396]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 11]
```diff
@@ -1429,7 +1437,8 @@ graph : :class:`~openturns.Graph`
 
 See Also
 --------
-computePDF, getMarginal, viewer.View, ResourceMap
+openturns.Distribution.computePDF
+openturns.Distribution.getMarginal
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:1441]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 12]
```diff
@@ -1473,7 +1482,8 @@ graph : :class:`~openturns.Graph`
 
 See Also
 --------
-computeLogPDF, getMarginal, viewer.View, ResourceMap
+openturns.Distribution.computeLogPDF
+openturns.Distribution.getMarginal
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:1486]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 13]
```diff
@@ -1537,7 +1547,7 @@ Only valid for univariate and bivariate distributions.
 
 See Also
 --------
-computePDF, viewer.View, ResourceMap
+openturns.Distribution.computePDF
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:1550]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 14]
```diff
@@ -1610,7 +1620,9 @@ Only valid for univariate and bivariate distributions.
 
 See Also
 --------
-computeLogPDF, viewer.View, ResourceMap
+openturns.Distribution.computeLogPDF
+openturns.viewer.View
+openturns.ResourceMap
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:1625]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 15]
```diff
@@ -1668,7 +1680,9 @@ In the case of bivariate distributions, defined by its CDF :math:`F` and its mar
 
 See Also
 --------
-computeQuantile, viewer.View, ResourceMap
+openturns.Distribution.computeQuantile
+openturns.viewer.View
+openturns.ResourceMap
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:1685]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 16]
```diff
@@ -1903,7 +1917,7 @@ Linv : :class:`~openturns.TriangularMatrix`
 
 See also
 --------
-getCholesky"
+openturns.Distribution.getCholesky"
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:1920]: I don't understand why the change is necessary here. The link already works.

File: python/src/DistributionImplementation_doc.i.in [hunk 17]
```diff
@@ -1931,7 +1945,7 @@ The inverse iso-probabilistic transformation is defined as follows:
 
 See also
 --------
-getIsoProbabilisticTransformation"
+openturns.Distribution.getIsoProbabilisticTransformation"
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:1948]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 18]
```diff
@@ -2005,7 +2019,9 @@ To be more specific:
 
 See also
 --------
-getInverseIsoProbabilisticTransformation, isElliptical, hasEllipticalCopula"
+openturns.Distribution.getInverseIsoProbabilisticTransformation
+openturns.Distribution.isElliptical
+openturns.Distribution.hasEllipticalCopula"
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:2024]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 19]
```diff
@@ -2363,7 +2379,10 @@ sample : :class:`~openturns.Sample`
 
 See Also
 --------
-getRealization, RandomGenerator, getSample, getSampleByQMC"
+openturns.Distribution.getRealization
+openturns.RandomGenerator
+openturns.Distribution.getSample
+openturns.Distribution.getSampleByQMC"
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:2385]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 20]
```diff
@@ -2386,7 +2405,10 @@ sample : :class:`~openturns.Sample`
 
 See Also
 --------
-getRealization, RandomGenerator, getSample, getSampleByInversion"
+openturns.Distribution.getRealization
+openturns.RandomGenerator
+openturns.Distribution.getSample
+openturns.Distribution.getSampleByInversion"
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:2411]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 21]
```diff
@@ -3054,7 +3076,7 @@ Only valid for univariate and bivariate distributions.
 
 See Also
 --------
-computeSurvivalFunction, viewer.View, ResourceMap
+openturns.Distribution.computeSurvivalFunction
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:3079]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 22]
```diff
@@ -3102,7 +3124,10 @@ graph : :class:`~openturns.Graph`
 
 See Also
 --------
-computeSurvivalFunction, getMarginal, viewer.View, ResourceMap
+openturns.Distribution.computeSurvivalFunction
+openturns.Distribution.getMarginal
+openturns.viewer.View
+openturns.ResourceMap
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:3130]: Same

File: python/src/DistributionImplementation_doc.i.in [hunk 23]
```diff
@@ -3147,7 +3172,8 @@ graph : :class:`~openturns.Graph`
 
 See Also
 --------
-computeSurvivalFunction, getMarginal, viewer.View, ResourceMap
+openturns.Distribution.computeSurvivalFunction
+openturns.Distribution.getMarginal
```
Comment (josephmure) [inline python/src/DistributionImplementation_doc.i.in:3176]: Same

File: python/src/EventSimulation_doc.i.in [hunk 24]
```diff
@@ -83,7 +81,10 @@ with a sample size :math:`> 2^{32}` by a combination of *blockSize* and
 
 see also
 --------
-setBlockSize, setMaximumOuterSampling, ResourceMap, SimulationResult"
+openturns.EventSimulation.setBlockSize
+openturns.EventSimulation.setMaximumOuterSampling
+openturns.ResourceMap
+openturns.SimulationResult"
```
Comment (josephmure) [inline python/src/EventSimulation_doc.i.in:87]: Same

File: python/src/FieldToPointFunctionalChaosAlgorithm_doc.i.in [hunk 25]
```diff
@@ -93,7 +93,9 @@ y : :class:`~openturns.Sample`
 
 See also
 --------
-FieldFunctionalChaosSobolIndices, KarhunenLoeveAlgorithm, FunctionalChaosAlgorithm
+openturns.experimental.FieldFunctionalChaosSobolIndices
+openturns.KarhunenLoeveAlgorithm
+openturns.FunctionalChaosAlgorithm
```
Comment (josephmure) [inline python/src/FieldToPointFunctionalChaosAlgorithm_doc.i.in:98]: Same

File: python/src/FittingTest_doc.i.in
Comment (josephmure) [inline python/src/FittingTest_doc.i.in:1]: Same in the whole file

File: python/src/GeneralizedExtremeValueValidation_doc.i.in [hunk 26]
```diff
@@ -14,7 +14,7 @@ sample : 2-d sequence of float
 
 See also
 --------
-GeneralizedExtremeValueFactory"
+openturns.GeneralizedExtremeValueFactory"
```
Comment (josephmure) [inline python/src/GeneralizedExtremeValueValidation_doc.i.in:17]: Same

File: python/src/HypothesisTest_doc.i.in
Comment (josephmure) [inline python/src/HypothesisTest_doc.i.in:1]: Same in the whole file

File: python/src/LatentVariableModel_doc.i.in [hunk 27]
```diff
@@ -54,7 +54,7 @@ rely on a constant functional basis: please use the :class:`~openturns.ConstantB
 
 See Also
 --------
-CovarianceModel
+openturns.CovarianceModel
```
Comment (josephmure) [inline python/src/LatentVariableModel_doc.i.in:57]: Same

File: python/src/LinearModelTest_doc.i.in
Comment (josephmure) [inline python/src/LinearModelTest_doc.i.in:1]: Same in the whole file

File: python/src/NormalityTest_doc.i.in
Comment (josephmure) [inline python/src/NormalityTest_doc.i.in:1]: Same in the whole file

File: python/src/PosteriorDistribution_doc.i.in [hunk 28]
```diff
@@ -12,9 +12,10 @@ conditional : :class:`~openturns.ConditionalDistribution`
 observations : 2-d sequence of float
     Observations :math:`x_i`.
 
-See also
+See Also
 --------
-ConditionalDistribution, MetropolisHastings
+openturns.ConditionalDistribution
+openturns.MetropolisHastings
```
Comment (josephmure) [inline python/src/PosteriorDistribution_doc.i.in:18]: same

File: python/src/RandomVectorImplementation_doc.i.in
Comment (josephmure) [inline python/src/RandomVectorImplementation_doc.i.in:1]: Same in the whole file

File: python/src/SmolyakExperiment_doc.i.in [hunk 29]
```diff
@@ -215,9 +215,9 @@ Therefore,
     - S_\ell(g)\right|
     = O \left(n_\ell^{-r} (\log(n_\ell))^{(d_x - 1)(r + 1)}\right).
 
-See also
+See Also
 --------
-WeightedExperiment
+openturns.WeightedExperiment
```
Comment (josephmure) [inline python/src/SmolyakExperiment_doc.i.in:220]: same

File: python/src/SmoothedUniformFactory_doc.i.in [hunk 30]
```diff
@@ -26,7 +26,8 @@ These are then used to initialize a numerical maximum likelihood step.
 
 See also
 --------
-DistributionFactory, SmoothedUniform"
+openturns.DistributionFactory
+openturns.SmoothedUniform"
```
Comment (josephmure) [inline python/src/SmoothedUniformFactory_doc.i.in:30]: Same

File: python/src/SpecFunc_doc.i.in
Comment (josephmure) [inline python/src/SpecFunc_doc.i.in:1]: Same in the whole file

File: python/src/SpectralModelFactoryImplementation_doc.i.in [hunk 31]
```diff
@@ -23,11 +16,10 @@ The bilateral spectral density function :math:`S : \Rset^n \rightarrow \mathcal{
...
 where :math:`\mathcal{H}^+(d) \in \mathcal{M}_d(\Cset)` is the set of *d*-dimensional positive definite hermitian matrices.
 
-Depending on the available data, we proceed differently :
-
-    -if the data correspond to  several independent  realizations of the  process, the estimation is done using the empirical estimator;
+Depending on the available data, we proceed differently:
 
-    - if the data correspond to one realization of the  process,  we suppose the process is ergodic to split the realization into several ones.
+- if the data correspond to several independent  realizations of the  process, the estimation is done using the empirical estimator;
+- if the data correspond to one realization of the  process,  we suppose the process is ergodic to split the realization into several ones.
```
Comment (josephmure) [inline python/src/SpectralModelFactoryImplementation_doc.i.in:22]:
```suggestion
- if the data correspond to several independent realizations of the  process, the estimation is done using the empirical estimator;
- if the data correspond to one realization of the process,  we suppose the process is ergodic to split the realization into several ones.
```

File: python/src/StudentCopula_doc.i.in [hunk 32]
```diff
@@ -31,7 +31,7 @@ for :math:`(u_1, \cdots, u_n) \in [0, 1]^n`
 
 See also
 --------
-Student, SklarCopula
+openturns.Student, openturns.SklarCopula
```
Comment (josephmure) [inline python/src/StudentCopula_doc.i.in:34]: same

File: python/src/UniformOverMesh_doc.i.in [hunk 33]
```diff
@@ -8,7 +8,7 @@ mesh : :class:`~openturns.Mesh`
 
 See also
 --------
-TruncatedOverMesh
+openturns.experimental.TruncatedOverMesh
```
Comment (josephmure) [inline python/src/UniformOverMesh_doc.i.in:11]: same

File: python/src/VisualTest_doc.i.in [hunk 34]
```diff
@@ -48,7 +48,8 @@ The Henry plot is a special case of the more general QQ-plot.
 
 See Also
 --------
-VisualTest.DrawQQplot, FittingTest.Kolmogorov
+openturns.VisualTest.DrawQQplot
+openturns.FittingTest.Kolmogorov
```
Comment (josephmure) [inline python/src/VisualTest_doc.i.in:52]: same

**Review by josephmure — COMMENTED** [review]
Great job, which will vastly improve the consistency of the documentation. I have a few questions but we should be able to merge this soon.

File: python/src/DistributionImplementation_doc.i.in (hunk 16 above)
Comment (jschueller) [reply python/src/DistributionImplementation_doc.i.in:1920]: I think its because of the distribution in experimental where inherited docstrings dont have the same module reference so they become invalid for these particular distributions

File: python/src/DistributionImplementation_doc.i.in (hunk 6 above)
Comment (jschueller) [reply python/src/DistributionImplementation_doc.i.in:826]: lets do that in another PR

File: python/doc/user_manual/base_objects.rst (hunk 4 above)
Comment (jschueller) [reply python/doc/user_manual/base_objects.rst:111]: the problem is that if we want to actually document the actual type returned we need these objects to be referenced somewhere, but else we could also say "sequence of XXX" or "list of XXX" for these

File: python/doc/examples/reliability_sensitivity/design_of_experiments/plot_smolyak_indices.py (hunk 2 above)
Comment (jschueller) [reply python/doc/examples/reliability_sensitivity/design_of_experiments/plot_smolyak_indices.py:36]: Maybe its the same, the problem was really the missing "experimental"

File: python/src/SpectralModelFactoryImplementation_doc.i.in [hunk 35]
```diff
@@ -23,11 +16,10 @@ The bilateral spectral density function :math:`S : \Rset^n \rightarrow \mathcal{
...
 where :math:`\mathcal{H}^+(d) \in \mathcal{M}_d(\Cset)` is the set of *d*-dimensional positive definite hermitian matrices.
 
-Depending on the available data, we proceed differently :
-
-    -if the data correspond to  several independent  realizations of the  process, the estimation is done using the empirical estimator;
+Depending on the available data, we proceed differently:
 
-    - if the data correspond to one realization of the  process,  we suppose the process is ergodic to split the realization into several ones.
+- if the data correspond to several independent realizations of the  process, the estimation is done using the empirical estimator;
+- if the data correspond to one realization of the process,  we suppose the process is ergodic to split the realization into several ones.
```
Comment (mbaudin47) [inline python/src/SpectralModelFactoryImplementation_doc.i.in:22]:
```suggestion
- if the data correspond to several independent realizations of the process, the estimation is done using the empirical estimator;
- if the data correspond to one realization of the process, we suppose the process is ergodic to split the realization into several ones.
```

**Comment by mbaudin47** [comment]
If you have run the tool, can you please share the output of the [linkcheck.html](https://www.sphinx-doc.org/en/master/_modules/sphinx/builders/linkcheck.html) tool? That would enable us to see how it looks like.

**Comment by mbaudin47** [comment]
You are right: there are many broken links which are detected by the tool. Here is an example from [plot_imh_python_distribution.html](http://openturns.github.io/openturns/latest/auto_calibration/bayesian_calibration/plot_imh_python_distribution.html):

image

**Figure 1.** A link, before the PR.

image

**Figure 2.** A link, after the PR.

File: python/src/TestResult_doc.i.in [hunk 36]
```diff
@@ -3,9 +3,9 @@
 
 .. warning::
 
-    Tests results are not intended to be created manually. They are returned
-    by the statistical tests implemented in the :mod:`~openturns.stattests`
-    subpackage. Constructor is therefore intentionally not documented.
+    Tests results are not intended to be created manually.
```
Comment (mbaudin47) [inline python/src/TestResult_doc.i.in:6]: As a side comment unrelated to this PR, the fact that `TestResult` is intentionally undocumented is a strange choice. There is no particular reason to prevent users from creating _any_ object.

**Review by mbaudin47 — CHANGES_REQUESTED** [review]
Thank you for this very significant improvement. Only minor changes may improve the content.

File: python/src/TestResult_doc.i.in (hunk 36 above)
Comment (jschueller) [reply python/src/TestResult_doc.i.in:6]: yes I agree

**Comment by jschueller** [comment]
> If you have run the tool, can you please share the output of the [linkcheck.html](https://www.sphinx-doc.org/en/master/_modules/sphinx/builders/linkcheck.html) tool? That would enable us to see how it looks like.

Its mostly nitpicky at play here for internal links, it just give an error or warning for missing refs,  but linkcheck is similar for external links

**Review by mbaudin47 — APPROVED** [review]

**Review by josephmure — APPROVED** [review]
